wordpress-mobile/WordPress-iOS
# GiphyService: Search and fetch thumbnails

Implements #9991.

This PR implements a basic `GiphyService` and associated classes (`DataSource`, `DataLoader`, `Pageable`, etc) – all of which mirror their StockPhotos counterparts. I've added a note to the Github project for later in the process to see if we can remove any duplication between these two areas.

The service is exposed in the Media Library section of the app. It currently allows searching for a term, paging results, fetching of thumbnails.

![giphy-12](https://user-images.githubusercontent.com/4780/44474175-f0e1cd00-a629-11e8-8b19-73acb2786fa3.gif)

Loading full resolution animated images for larger previews currently isn't supported.

**To test:**

* Check the code. As I mentioned, a lot of it is verbatim from the Stock Photos implementation.
* Ensure you have an up-to-date `.wpcom_app_credentials` file.
* Build and run, and navigate to the **Media** section for a site.
* Tap **+** and choose **Giphy** from the menu.
* Enter a search term, and check results are returned.
* Scroll down and check the results are paged.
* To verify that results are paged _correctly_ I tried to find a search term with a limited number of results. The best I could come up with is to do the following:
  * In the `search` method of `GiphyService`, remove the `rating:` parameter from the call to `GiphyService.giphy.search` – by default we're specifying a rating of G for images, but that doesn't return quite enough results for the term I found.
  * Build and run and search for `eurogamer` – check that there are 50 results displayed.
